app/src/main/java/com/github/se/polyfit/viewmodel/meal/MealViewModel.kt

```diff
@@ -3,62 +3,46 @@ package com.github.se.polyfit.viewmodel.meal
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.map
 import com.github.se.polyfit.data.remote.firebase.MealFirebaseRepository
 import com.github.se.polyfit.model.ingredient.Ingredient
 import com.github.se.polyfit.model.meal.Meal
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
 
-class MealViewModel(
-    private val userId: String,
-    firebaseID: String = "",
-    var initialMeal: Meal? = null,
-    private val mealRepo: MealFirebaseRepository = MealFirebaseRepository(userId)
-) : ViewModel() {
-  // after friday use hilt dependency injection to make code cleaner, for now i guess this is ok
-  private val _meal: MutableLiveData<Meal> = MutableLiveData(null)
-  val meal: LiveData<Meal> = _meal
+@HiltViewModel
+class MealViewModel @Inject constructor(private val mealRepo: MealFirebaseRepository) :
+    ViewModel() {
 
-  // Todo: If find a way to import Transformations, can use that to prevent duplicating updates
-  private val _isComplete: MutableLiveData<Boolean> = MutableLiveData(false)
-  val isComplete: LiveData<Boolean> = _isComplete
+  private val _meal: MutableLiveData<Meal> = MutableLiveData(Meal.default())
+  val meal: LiveData<Meal> = _meal // Expose _meal as an immutable LiveData
 
-  init {
-    if (firebaseID.isNotEmpty()) {
-      mealRepo.getMeal(firebaseID).addOnCompleteListener {
-        if (it.isSuccessful) {
-          _meal.value = it.result
-          _isComplete.value = it.result?.isComplete() ?: false
-        }
-      }
-    } else {
-      _meal.value = initialMeal?.copy() ?: Meal.default()
-      _isComplete.value = _meal.value?.isComplete() ?: false
-    }
+  private val _isComplete: LiveData<Boolean> = _meal.map { it?.isComplete() ?: false }
```

Comment:
Doesn't `Meal` class already have a `isComplete` function? Like why are we making it into a map? Or livedata here works differently?

Reply:
Live data is better when used in the UI, because an observer can be placed on it, allowing the UI to react directly to it.